numba/cuda/tests/cudapy/test_intrinsics.py

```diff
@@ -394,6 +399,70 @@ def test_round_f8(self):
             compiled[1, 1](ary, i)
             self.assertEquals(ary[0], round(i))
 
+    def test_round_to_f4(self):
+        compiled = cuda.jit("void(float32[:], float32, int32)")(simple_round_to)
+        ary = np.zeros(1, dtype=np.float32)
+        vals = np.random.random(32).astype(np.float32)
+        np.concatenate((vals, np.array([np.inf, -np.inf, np.nan])))
+        digits = (
+            # Common case branch of round_to_impl
+            -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5,
+            # The algorithm currently implemented can only round to 13 digits
+            # with single precision. Note that this doesn't trigger the
+            # "overflow safe" branch of the implementation, which can only be
+            # hit when using double precision.
+            13
+        )
+        for val, ndigits in itertools.product(vals, digits):
+            with self.subTest(val=val, ndigits=ndigits):
+                compiled[1, 1](ary, val, ndigits)
+                self.assertPreciseEqual(ary[0], round(val, ndigits),
+                                        prec='single')
+
+    def test_round_to_f4_halfway(self):
+        # Presently not passing
```

Comment:
This is passing?!

Reply:
Accidentally left in this comment from an earlier commit - it is indeed passing now :-)